### ISSUE TITLE
`CometLogger` and `SaveConfigCallback`

### ISSUE THREAD

**Issue by samvanstroud** [PyTorchLightning/pytorch-lightning#15085]
### Bug description

Similar, but not identical to https://github.com/Lightning-AI/lightning/issues/12803

The default `SaveConfigCallback` is not well compatible with the `CometLogger` (and perhaps other loggers). 

Given
```yaml
  logger:
    class_path: pytorch_lightning.loggers.CometLogger
    init_args:
      save_dir: logs
      project_name: project
```
`SaveConfigCallback` saves the `config.yaml` to `logs/`, while the logger writes checkpoints to `logs/project/<id>/checkpoints`.

This means that upon the second run, `SaveConfigCallback` complains that the `config.yaml` already exists and refuses to overwrite it. Instead the `config.yaml` should be stored alongside the checkpoints. This is possible with some basic modifications to the `SaveConfigCallback`, i.e. taking the save dir from the `trainer.logger` properties, instead of from [`trainer.log_dir`](https://github.com/Lightning-AI/lightning/blob/8715cd034615f7d2c631cd0724173026134f8e08/src/pytorch_lightning/cli.py#L215). However, I don't know if this can be made consistent across all loggers.

**Comment by awaelchli** [comment]
@samvanstroud Yes, we noticed this too and this will be entirely solved by #14188 which we already started working on. Trainer.log_dir will be returning a consistent path to the experiment dir for all loggers and the LightningCLI will be able to properly store the config file there, as you said, where it belongs.

**Comment by samvanstroud** [comment]
Great thanks @awaelchli, closing this in that case
